app/assets/javascripts/discourse/app/components/decorated-html.gjs

```diff
@@ -0,0 +1,79 @@
+import Component from "@glimmer/component";
+import { untrack } from "@glimmer/validator";
+import { htmlSafe, isHTMLSafe } from "@ember/template";
+import { TrackedArray } from "@ember-compat/tracked-built-ins";
+import helperFn from "discourse/helpers/helper-fn";
+
+const detachedDocument = document.implementation.createHTMLDocument("detached");
+
+/**
+ * Reactively renders cooked HTML with decorations applied.
+ */
+export default class DecoratedHtml extends Component {
```

Comment:
Wonder if it should have "cooked" in the name

Reply:
Yeah I struggled with the name here. I don't really want to call it "DecoratedCooked", because that's too similar to our `decorateCooked` API, which only applies to posts. This component will be used more widely, and may even be used for things which are not "cooked" post/message content.